src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -48,19 +55,25 @@
     private final FireStoreDao fileDao;
     private final ApplicationContext applicationContext;
     private final ConfigurationService configurationService;
+    private final ExecutorService executor;
+    private final PerformanceLogger performanceLogger;
 
     @Autowired
     public GcsPdao(
         GcsProjectFactory gcsProjectFactory,
         DataLocationService dataLocationService,
         FireStoreDao fileDao,
         ApplicationContext applicationContext,
-        ConfigurationService configurationService) {
+        ConfigurationService configurationService,
+        @Qualifier("performanceThreadpool") ExecutorService executor,
```

Comment:
Huh, I've never had to use @Qualifier before. This is needed because you wanted to call it `executor`?

Reply:
Yeah, I put it in pre-emptively (perhaps that's confusing) since we may potentially have multiple Executor services.  This let's us get this one specifically.  I can remove for now since we probably don't need it yet

Reply:
I wouldn't change it, since we could get weird errors if we add more Executor services in the future.
Since I hadn't seen it before, I wanted to make sure I understood what it was.

Reply:
👍